From 3167d4dfb82aa74fcf41c755d6c9bc7a3401bfea Mon Sep 17 00:00:00 2001
From: Timo Sandmann <mail@timosandmann.de>
Date: Sun, 8 Jan 2017 18:24:20 +0100
Subject: [PATCH] Fixes #38

Use correct type cast and tuple extracting for pointer to transfered data tuples
---
 include/boost/context/execution_context_v2.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boost/context/execution_context_v2.hpp b/boost/context/execution_context_v2.hpp
index 33b9bda..bbd4eb1 100644
--- a/boost/context/execution_context_v2.hpp
+++ b/boost/context/execution_context_v2.hpp
@@ -103,7 +103,7 @@ class record {
 
     transfer_t run( transfer_t t) {
         Ctx from{ t.fctx };
-        typename Ctx::args_tpl_t args = std::move( * static_cast< typename Ctx::args_tpl_t * >( t.data) );
+        typename Ctx::args_tpl_t args = std::move( std::get<1>( * static_cast< std::tuple< std::exception_ptr, typename Ctx::args_tpl_t > * >( t.data) ) );
         auto tpl = std::tuple_cat(
                     params_,
                     std::forward_as_tuple( std::move( from) ),
